### PR TITLE
[mono] Some UnsafeAccessorAttribute support for methods and constructors

### PR DESCRIPTION
**Pull request by lambdageek**
Just the simple cases so far that don't require considering custom modifiers.

Contributes to #86040

### CODE REVIEW

**Comment by lambdageek** [comment]
I would really like to also support `Verify_InvalidUseUnsafeAccessor` in this PR:

https://github.com/dotnet/runtime/blob/9915e9825edf174df73bc6db9ab59f167e92ab3c/src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.cs#L456-L478

**Comment by lambdageek** [comment]
I moved the `CallAmbiguousMethod` test to a separate testcase - in general the custom modifiers stuff in Mono is not done.  Only exact matches are working.

**Comment by lambdageek** [comment]
Also need to try out the new tests from https://github.com/dotnet/runtime/pull/88268

**Comment by lambdageek** [comment]
Pushed some fixes to match the behavior expected by the new tests from https://github.com/dotnet/runtime/pull/88268

**Comment by lambdageek** [comment]
There will be merge conflicts with #88268 the resolution is on https://github.com/lambdageek/runtime/tree/unsafe-accessor-kind-merge-vitek

File: src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.cs [hunk 1]
```diff
@@ -429,6 +412,17 @@ public static void Verify_InvalidTargetUnsafeAccessor()
         [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name=UserDataClass.MethodPointerName)]
         extern static string CallPointerMethod(UserDataClass d, delegate* unmanaged[Stdcall]<void> fptr);
 
+    }
+
```
Comment (AaronRobinsonMSFT) [inline src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.cs:416]:
```suggestion
```

File: src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.cs [hunk 2]
```diff
@@ -429,6 +412,17 @@ public static void Verify_InvalidTargetUnsafeAccessor()
...
+
+    [Fact]
+    [ActiveIssue("https://github.com/dotnet/runtime/issues/86040", TestRuntimes.Mono)]
+    public static void Verify_InvalidTargetUnsafeAccessorAmbiguousMatch()
+    {
+        Assert.Throws<AmbiguousMatchException>(
+            () => CallAmbiguousMethod(CallPrivateConstructorClass(), null));
+
```
Comment (AaronRobinsonMSFT) [inline src/tests/baseservices/compilerservices/UnsafeAccessors/UnsafeAccessorsTests.cs:424]:
```suggestion
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by fanyang-mono — APPROVED** [review]
LGTM!

**Pull request merged**